interp/src/debugger/cidr.rs

```diff
@@ -87,37 +89,32 @@ impl<'a> Debugger<'a> {
                 }
                 Command::Empty => {}
                 Command::Display => {
-                    let states = component_interpreter.get_env();
-                    println!(
-                        "{}",
-                        if states.len() == 1 {
-                            states[0].state_as_str()
-                        } else {
-                            "There are mutliple states".into()
-                        }
-                    );
+                    let state = component_interpreter.get_env();
+                    println!("{}", state.state_as_str());
                 }
                 Command::PrintCell(cell) => {
-                    let env: Vec<_> = component_interpreter
-                        .get_env()
-                        .into_iter()
-                        .map(|x| (x, x.get_cell(&cell)))
-                        .collect();
-                    if env.iter().any(|(_, x)| x.len() > 1) {
-                        println!(
-                            "{}Unable to print. '{}' is ambiguous",
-                            SPACING, &cell
-                        )
-                    } else if env.iter().all(|(_, x)| x.is_empty()) {
-                        println!(
-                            "{}Unable to print. No cell named '{}'",
-                            SPACING, &cell
-                        )
-                    } else {
-                        for (state, cells) in env {
-                            if let Some(cell_ref) = cells.first() {
-                                print_cell(cell_ref, state)
-                            }
+                    let cells: Vec<_> =
+                        component_interpreter.get_env().get_cells(&cell);
+
+                    match cells.len() {
+                        0 => {
+                            println!(
+                                "{}Unable to print. No cell named '{}'",
+                                SPACING, &cell
+                            )
+                        }
+                        1 => {
+                            let cell_ref = &cells[0];
+                            print_cell(
+                                cell_ref,
+                                &component_interpreter.get_env(),
+                            );
+                        }
+                        _ => {
```

Comment:
Is this case possible? The frontend will disallow defining multiple cells with the same name.

Reply:
It is possible because this method searches for cells with a matching name on all components. This should be changed pending rewriting print in general

Reply:
Note that this isn't actually useful since components other than the main component are necessarily instanced, but I don't know that it's worth fixing right now since this code is going to be removed soon anyway.